lldb/source/Host/posix/HostInfoPosix.cpp

```diff
@@ -29,6 +29,10 @@
 #include <sys/utsname.h>
 #include <unistd.h>
 
+#ifdef __ANDROID__
+#include <android/api-level.h>
```

Comment:
what's the build failure you're seeing? this _should_ be included "for free" by all of the #includes above (via <sys/cdefs.h>, which they should all drag in).

of course, if you're aiming for a style guide "explicitly include what you use" kind of thing, that's fine, but -- other than for style reasons -- you shouldn't ever _need_ this...

Reply:
It sounds like it's mainly for "style", but given what you've said, maybe that shouldn't be the style we should use (is it possible those includes were necessary 10 years ago?) I think this code would look much nicer if we could assume that __ANDROID_API__ is always defined (at least that, like, we could assume that any header which includes the definition of a C function we're about to call, also includes the definition of that macro -- which I guess must be true as otherwise it wouldn't be able to annotate the declaration)

Reply:
> is it possible those includes were necessary 10 years ago?

yeah, one reason i'm so confidently asserting that you shouldn't need this now is because i know we had a big cleanup to ensure that we hadn't missed any :-)

> we could assume that any header which includes the definition of a C function we're about to call, also includes the definition of that macro -- which I guess must be true as otherwise it wouldn't be able to annotate the declaration

correct.